boards/feather_m0/examples/async_dmac.rs

```diff
@@ -0,0 +1,73 @@
+//! This example shows a safe API to
+//! execute a memory-to-memory DMA transfer
+
+#![no_std]
+#![no_main]
+
+use defmt_rtt as _;
+use panic_probe as _;
+
+atsamd_hal::bind_interrupts!(struct Irqs {
+    DMAC => atsamd_hal::dmac::InterruptHandler;
+});
+
+use bsp::hal;
+use bsp::pac;
+use feather_m0 as bsp;
+use hal::{
+    clock::GenericClockController,
+    dmac::{DmaController, PriorityLevel, TriggerAction, TriggerSource},
+};
+
+#[embassy_executor::main]
+async fn main(_s: embassy_executor::Spawner) {
+    let mut peripherals = pac::Peripherals::take().unwrap();
+    let _core = pac::CorePeripherals::take().unwrap();
+
+    let _clocks = GenericClockController::with_external_32kosc(
+        peripherals.GCLK,
+        &mut peripherals.PM,
+        &mut peripherals.SYSCTRL,
+        &mut peripherals.NVMCTRL,
+    );
+
+    // Initialize DMA Controller
+    let dmac = DmaController::init(peripherals.DMAC, &mut peripherals.PM);
+
+    // Turn dmac into an async controller
+    let mut dmac = dmac.into_future(crate::Irqs);
+    // Get individual handles to DMA channels
+    let channels = dmac.split();
+
+    // Initialize DMA Channel 0
+    let mut channel = channels.0.init(PriorityLevel::LVL0);
+
+    let mut source = [0xff; 500];
+    let mut dest = [0x0; 500];
+
+    defmt::info!(
+        "Launching a DMA transfer.\n\tSource: {}\n\tDestination: {}",
+        &source,
+        &dest
+    );
+
+    channel
+        .transfer_future(
+            &mut source,
+            &mut dest,
+            TriggerSource::DISABLE,
+            TriggerAction::BLOCK,
+        )
+        .await
+        .unwrap();
+
+    defmt::info!(
+        "Finished DMA transfer.\n\tSource: {}\n\tDestination: {}",
+        &source,
+        &dest
+    );
+
+    loop {
+        cortex_m::asm::wfi();
```

Comment:
should this be `core::hint::spin_loop`? genuinely not sure just curious

Reply:
I don't think it really matters; the idea is just that WFI should help reduce power consumption while idle. I can just as well change it to a `spin_loop` if needed.

Reply:
yeah makes sense. It doesn't matter and the power saving is good. I wonder if we should have both? but meh

Reply:
Nah, the spin_loop hint is only necessary to tell the compiler not to optimize the loop out. As soon as there's at least one instruction in there, it's no longer necessary.